utils/unittest/entity.go

```diff
@@ -59,26 +60,26 @@ func RequireEntityNonMalleable(t *testing.T, entity flow.IDEntity, ops ...Mallea
 // The MalleabilityChecker performs a recursive check of all fields of the entity and ensures that changing any field will change
 // the ID of the entity. By default, the MalleabilityChecker uses pre-defined generators for each basic golang type, which return
 // a random value, to modify the entity's field values. However, the MalleabilityChecker can be customized, by providing custom
-// types and their generators.
+// generators for specific types or specific fields.
 //
-// The caller must provide a properly instantiated entity struct, which serves a template for further modification.
-// Input entities must have all non-nil and non-empty slice/map fields, otherwise `Check` will return an error.
+// The caller can provide a loosely instantiated entity struct, which serves a template for further modification.
+// If checker encounters a field that is nil, it will create a new instance of the field and continue the check.
+// If checker encounters a nil/empty slice or map, it will create a new instance of the slice/map, insert a value and continue the check.
 // In rare cases, a type may have a different ID computation depending on whether a field is nil.
-// In such cases, we can use the `malleability:"optional"` struct tag to skip malleability checks when the field is nil.
+// In such cases, we can use the WithPinnedField option to skip malleability checks on this field.
 //
 // This checker heavily relies on generation of random values for the fields based on their type. All types are split into three categories:
-//  1. structures (generateCustomFlowValue)
+//  1. structures, primitives, slices, arrays, maps (generateRandomReflectValue)
 //  2. interfaces (generateInterfaceFlowValue)
-//  3. primitives, slices, arrays, maps (generateRandomReflectValue)
 //
 // Checker knows how to deal with each of the categories and generate random values for them.
 // There are two ways to handle types not natively recognized byt he MalleabilityChecker:
 //  1. User can provide a custom type generator for the type using WithTypeGenerator option.
-//  2. User can extend the checker with new type handling.
+//  2. User can provide a custom generator for the field using WithFieldGenerator option.
 //
-// It is recommended to use the second option if type is used in multiple places and general enough. For other cases
-// it is better to use the first option. Mind that the first option overrides any type handling that is embedded in the checker.
-// This is very useful for cases where the field is context-sensitive, and we cannot generate a completely random value.
+// It is recommended to use the first option if type is used in multiple places and general enough. For other cases
+// it is better to use the second option.
```

Comment:
```suggestion
// It is recommended to use the first option if type is used in multiple places and general enough.
// Matching by type (instead of field name) is less selective, by covering all fields of the given type.
```